dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -646,19 +662,20 @@ public SolrQueryResponse search(DataverseRequest dataverseRequest, List<Datavers
              * we'll build a hashmap
              */
             for (DatasetFieldType datasetField : datasetFields) {
-                String solrFieldNameForDataset = datasetField.getSolrField().getNameFacetable();
+                SolrField dsfSolrField = solrFieldFactory.getSolrField(datasetField.getName(),
+                                                                       datasetField.getFieldType(),
+                                                                       datasetField.isThisOrParentAllowsMultipleValues(),
+                                                                       datasetField.isFacetable());
+                String solrFieldNameForDataset = dsfSolrField.getNameFacetable();
                 String friendlyName = datasetField.getDisplayName();
-                if (solrFieldNameForDataset != null && facetField.getName().endsWith(datasetField.getTmpNullFieldTypeIdentifier())) {
-                    // give it the non-friendly name so we remember to update the reference data script for datasets
-                    facetCategory.setName(facetField.getName());
-                } else if (solrFieldNameForDataset != null && facetField.getName().equals(solrFieldNameForDataset)) {
+                if (solrFieldNameForDataset != null && facetField.getName().equals(solrFieldNameForDataset)) {
```

Comment:
Why do we change the logic of this statement (in refactor task)?

Reply:
Actually it is not logic change it is rather cleanup, the column fieldtype is (nullable = false) so the is no way to execute that logic (it was probably there due to legacy reasons).